go/common/headers.go

```diff
@@ -59,12 +63,13 @@ type HeaderWithTxHashes struct {
 	TxHashes []TxHash
 }
 
-// Hash returns the block hash of the header, which is simply the keccak256 hash of its
+// CalcHash returns the block hash of the header, which is simply the keccak256 hash of its
 // RLP encoding excluding the signature.
-func (h *Header) Hash() L2RootHash {
+func (h *Header) CalcHash() L2RootHash {
 	cp := *h
 	cp.R = nil
 	cp.S = nil
+	cp.Hash = common.Hash{}
```

Comment:
Worth a comment here to say you always use a copy with a zeroed hash to avoid circular calculation.

Reply:
Fixed